File: app/src/main/java/com/example/messengerandroid/service/ApiService.java
package com.example.messengerandroid.service;

import com.example.messengerandroid.model.Character;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

import java.util.List;

public interface ApiService {
    @GET("characters")
    Call<List<Character>> getCharacters(@Query("page") int page, @Query("pageSize") int pageSize);
}

File: app/src/main/java/com/example/messengerandroid/service/ApiService.kt
package com.example.messengerandroid.service

import com.example.messengerandroid.db.Character
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface ApiService {
    @GET("characters")
    fun getCharacters(
        @Query("page") page: Int,
        @Query("pageSize") pageSize: Int
    ): Call<List<Character>>
}